docs/excel/custom-functions-naming.md

```diff
@@ -0,0 +1,38 @@
+---
+ms.date: 02/08/2019
+description: Learn requirements for Excel custom functions' names and avoid common naming pitfalls.
+title: Naming guidelines for custom functions in Excel (preview)
+localization_priority: Normal
+---
+# Naming guidelines
+
+A custom function is identified by an `id` and `name` property in the JSON metadata file. The function `id` is used to uniquely identify custom functions in your JavaScript code. The function `name` is used as the display name that appears to a user in Excel. A function name can differ from the function ID, such as for localization purposes. But in general it should stay the same as the ID if there is no compelling reason for them to differ.
+
+Function names and function IDs share some common requirements:
+
+- They must only use alphanumeric characters (including Unicode), the numbers zero through nine, underscores, and periods.
+
+- They must start with a letter and have a minimum limit of three characters.
+
+Most function names for built-in Excel functions appear in uppercase letters (such as `SUM`). As a best practice, we recommend to use uppercase for your custom function names so they appear consistent for users.
```

Comment:
```suggestion
Excel uses uppercase letters for built-in function names (such as `SUM`). Therefore, consider using uppercase letters for your custom function names and function IDs as a best practice.
```

Reply:
Hmmmm, on the other hand, if people DO want to use a naming convention like Hungarian, lower camel case is very useful, too. Do we really want to suggest this as a best practice?

Reply:
@lindalu-MSFT  Yes, all functions in Excel appear to users as uppercase (when you type formulas in a cell), so you'll create a more consistent experience by naming your new custom functions also as uppercase.

Reply:
Real coders will still ignore this and use their naming conventions.